File: _fix_topup.py
import os

path = os.path.join(os.path.dirname(__file__), 'app', 'supplier', 'dashboard', 'page.tsx')
with open(path, 'r', encoding='utf-8') as f:
    content = f.read()

# 1. Replace: Restock priority line
old1 = '`Restock priority: "${b.name}" led today with ${b.quantity} units sold (${Number(b.total ?? 0).toLocaleString()} RWF).`'
new1 = '`${ui.restockPriority} "${b.name}" ${ui.ledTodayWith} ${b.quantity} ${ui.unitsSold} (${Number(b.total ?? 0).toLocaleString()} RWF).`'
assert old1 in content, "Could not find Restock priority line"
content = content.replace(old1, new1)

# 2. Replace: Runner-up line prefix
old2 = '`Runner-up: "${b.name}"'
new2 = '`${ui.runnerUp} "${b.name}"'
assert old2 in content, "Could not find Runner-up line"
content = content.replace(old2, new2)

# 3. Replace: Low stock bullet
old3 = '`Low stock: ${nm}'
new3 = '`${ui.lowStockColon} ${nm}'
assert old3 in content, "Could not find Low stock line"
content = content.replace(old3, new3)

old3b = 'only ${p.stock} left. Top up before it runs out.`'
new3b = '${ui.only} ${p.stock} ${ui.left}. ${ui.topUpBeforeRunsOut}`'
assert old3b in content, "Could not find 'only X left' part"
content = content.replace(old3b, new3b)

# 4. Replace: ordersRecordedNoLines push
old4 = '''      bullets.push(
        "Orders are recorded for today, but line items were empty. Deploy the latest backend (SellerOrdersServlet fix) and refresh \u2014 best-selling names will appear here.",
      );'''
new4 = '      bullets.push(ui.ordersRecordedNoLines);'
assert old4 in content, "Could not find ordersRecordedNoLines push"
content = content.replace(old4, new4)

# 5. Replace: fulfillOrdersTip push
old5 = '''      bullets.push(
        "Fulfill orders and keep fast movers in stock \u2014 tailored tips appear here from your live sales and inventory.",
      );'''
new5 = '      bullets.push(ui.fulfillOrdersTip);'
assert old5 in content, "Could not find fulfillOrdersTip push"
content = content.replace(old5, new5)

# 6. Replace: useMemo deps
old6 = '  }, [analytics, supplierProducts]);'
new6 = '  }, [analytics, supplierProducts, ui]);'
assert old6 in content, "Could not find useMemo deps"
content = content.replace(old6, new6)

with open(path, 'w', encoding='utf-8') as f:
    f.write(content)

print("All topUpSaleBullets replacements applied successfully!")

File: _fix_topup.mjs
import { readFileSync, writeFileSync } from 'fs';
import { join, dirname } from 'path';
import { fileURLToPath } from 'url';

const __dirname = dirname(fileURLToPath(import.meta.url));
const filePath = join(__dirname, 'app', 'supplier', 'dashboard', 'page.tsx');
let content = readFileSync(filePath, 'utf-8');

const replacements = [
  // 1. Restock priority line
  [
    '`Restock priority: "${b.name}" led today with ${b.quantity} units sold (${Number(b.total ?? 0).toLocaleString()} RWF).`',
    '`${ui.restockPriority} "${b.name}" ${ui.ledTodayWith} ${b.quantity} ${ui.unitsSold} (${Number(b.total ?? 0).toLocaleString()} RWF).`',
  ],
  // 2. Runner-up line prefix
  [
    '`Runner-up: "${b.name}"',
    '`${ui.runnerUp} "${b.name}"',
  ],
  // 3. Low stock bullet prefix
  [
    '`Low stock: ${nm}',
    '`${ui.lowStockColon} ${nm}',
  ],
  // 3b. "only X left..." part
  [
    'only ${p.stock} left. Top up before it runs out.`',
    '${ui.only} ${p.stock} ${ui.left}. ${ui.topUpBeforeRunsOut}`',
  ],
  // 4. ordersRecordedNoLines (em-dash is U+2014)
  [
    `"Orders are recorded for today, but line items were empty. Deploy the latest backend (SellerOrdersServlet fix) and refresh \u2014 best-selling names will appear here."`,
    'ui.ordersRecordedNoLines',
  ],
  // 5. fulfillOrdersTip (em-dash is U+2014)
  [
    `"Fulfill orders and keep fast movers in stock \u2014 tailored tips appear here from your live sales and inventory."`,
    'ui.fulfillOrdersTip',
  ],
  // 6. useMemo deps
  [
    '}, [analytics, supplierProducts]);',
    '}, [analytics, supplierProducts, ui]);',
  ],
];

let count = 0;
for (const [old, rep] of replacements) {
  if (content.includes(old)) {
    content = content.replace(old, rep);
    count++;
    console.log(`OK: replaced "${old.slice(0, 40)}..."`);
  } else {
    console.log(`SKIP (not found): "${old.slice(0, 60)}..."`);
  }
}

writeFileSync(filePath, content, 'utf-8');
console.log(`\nDone: ${count}/${replacements.length} replacements applied.`);
